packages/core/src/main/conversation/message/MessageService.ts

```diff
@@ -32,139 +32,108 @@ import {
   QualifiedUserClients,
   UserClients,
 } from '@wireapp/api-client/src/conversation';
-import {Decoder, Encoder} from 'bazinga64';
+import {Encoder} from 'bazinga64';
 
+import {encryptAsset} from '../../cryptography/AssetCryptography.node';
 import {CryptographyService} from '../../cryptography';
-import {QualifiedId, QualifiedUserPreKeyBundleMap} from '@wireapp/api-client/src/user';
+import {QualifiedId, QualifiedUserPreKeyBundleMap, UserPreKeyBundleMap} from '@wireapp/api-client/src/user';
+import {MessageBuilder} from './MessageBuilder';
+import {GenericMessage} from '@wireapp/protocol-messaging';
+import {GenericMessageType} from '..';
+import {flattenUserClients, flattenQualifiedUserClients} from './UserClientsUtil';
 
-type ClientMismatchError = AxiosError<ClientMismatch>;
+type ClientMismatchError = AxiosError<ClientMismatch | MessageSendingStatus>;
 
 export class MessageService {
   constructor(private readonly apiClient: APIClient, private readonly cryptographyService: CryptographyService) {}
 
-  public async sendOTRMessage(
+  /**
+   * Sends a message to a non-federated backend.
+   * @param {string} sendingClientId
+   * @param {UserClients|UserPreKeyBundleMap} recipients
+   * @param {Uint8Array} plainText
+   * @param {any} options:{conversationId?:string;reportMissing?:boolean;sendAsProtobuf?:boolean;onClientMismatch?:(mismatch:ClientMismatch
+   * @returns {any}
+   */
+  public async sendMessage(
```

Comment:
now there are only 2 public methods in the MessageService:
- sendMessage
- sendFederatedMessage

All the routing to (`sendExternal`, `sendProtobuf`, `sendRegular`...) is done internally and simplify the work for the consumer (as you can see, the `ConversationService` has been trimmed from this kind of logic)